Fix failing test on go1.13

_Sometimes_ one of the tests fails. Find out the root cause and fix. 
```
=== RUN   TestPubKeyVerify/Secp256k1
--- FAIL: TestPubKeyVerify (0.83s)
    --- PASS: TestPubKeyVerify/Secp256r1 (0.00s)
    --- FAIL: TestPubKeyVerify/Secp256k1 (0.83s)
        sign_verify_test.go:41: 
            	Error Trace:	sign_verify_test.go:41
            	Error:      	Should be true
            	Test:       	TestPubKeyVerify/Secp256k1
```

Maybe the same issue (go1.12):
```
--- FAIL: TestECDSAVerify (0.02s)
    logger.go:130: 2020-07-24T07:38:10.564Z	INFO	no storage version found! creating genesis block
    --- PASS: TestECDSAVerify/success (0.00s)
    --- PASS: TestECDSAVerify/signed_interop_item (0.00s)
    --- PASS: TestECDSAVerify/signed_script_container (0.00s)
    --- PASS: TestECDSAVerify/missing_arguments (0.00s)
    --- PASS: TestECDSAVerify/invalid_signature (0.00s)
    --- FAIL: TestECDSAVerify/invalid_public_key (0.00s)
        interop_neo_test.go:167: 
            	Error Trace:	interop_neo_test.go:167
            	            				interop_neo_test.go:214
            	Error:      	An error is expected but got nil.
            	Test:       	TestECDSAVerify/invalid_public_key
```

--- a/pkg/core/interop/crypto/ecdsa_test.go
+++ b/pkg/core/interop/crypto/ecdsa_test.go
@@ -97,16 +97,26 @@ func TestCHECKMULTISIGGood(t *testing.T) {
 	t.Run("12_9", func(t *testing.T) { testCHECKMULTISIGGood(t, 12, []int{0, 1, 4, 5, 6, 7, 8, 9}) })
 }
 
-func testCHECKMULTISIGBad(t *testing.T, n int, ik, is []int) {
+func testCHECKMULTISIGBad(t *testing.T, isErr bool, n int, ik, is []int) {
 	v := initCHECKMULTISIGVM(t, n, ik, is)
 
+	if isErr {
+		require.Error(t, v.Run())
+		return
+	}
 	require.NoError(t, v.Run())
 	assert.Equal(t, 1, v.Estack().Len())
 	assert.False(t, v.Estack().Pop().Bool())
 }
 
 func TestCHECKMULTISIGBad(t *testing.T) {
-	t.Run("1_1 wrong signature", func(t *testing.T) { testCHECKMULTISIGBad(t, 2, []int{0}, []int{1}) })
-	t.Run("3_2 wrong order", func(t *testing.T) { testCHECKMULTISIGBad(t, 3, []int{0, 2}, []int{2, 0}) })
-	t.Run("3_2 duplicate sig", func(t *testing.T) { testCHECKMULTISIGBad(t, 3, nil, []int{0, 0}) })
+	t.Run("1_1 wrong signature", func(t *testing.T) { testCHECKMULTISIGBad(t, false, 2, []int{0}, []int{1}) })
+	t.Run("3_2 wrong order", func(t *testing.T) { testCHECKMULTISIGBad(t, false, 3, []int{0, 2}, []int{2, 0}) })
+	t.Run("3_2 duplicate sig", func(t *testing.T) { testCHECKMULTISIGBad(t, false, 3, nil, []int{0, 0}) })
+	t.Run("1_2 too many signatures", func(t *testing.T) { testCHECKMULTISIGBad(t, true, 2, []int{0}, []int{0, 1}) })
+	t.Run("gas limit exceeded", func(t *testing.T) {
+		v := initCHECKMULTISIGVM(t, 1, []int{0}, []int{0})
+		v.GasLimit = ECDSAVerifyPrice - 1
+		require.Error(t, v.Run())
+	})
 }

--- a/pkg/core/interop_neo_test.go
+++ b/pkg/core/interop_neo_test.go
@@ -213,6 +213,12 @@ func TestECDSAVerify(t *testing.T) {
 		pub[0] = 0xFF // invalid prefix
 		runCase(t, true, false, sign, pub, msg)
 	})
+
+	t.Run("invalid message", func(t *testing.T) {
+		sign := priv.Sign(msg)
+		runCase(t, false, false, sign, priv.PublicKey().Bytes(),
+			stackitem.NewArray([]stackitem.Item{stackitem.NewByteArray(msg)}))
+	})
 }
 
 func TestRuntimeEncode(t *testing.T) {

--- a/pkg/core/interop_neo_test.go
+++ b/pkg/core/interop_neo_test.go
@@ -210,7 +210,7 @@ func TestECDSAVerify(t *testing.T) {
 	t.Run("invalid public key", func(t *testing.T) {
 		sign := priv.Sign(msg)
 		pub := priv.PublicKey().Bytes()
-		pub = pub[10:]
+		pub[0] = 0xFF // invalid prefix
 		runCase(t, true, false, sign, pub, msg)
 	})
 }

--- a/pkg/crypto/keys/private_key.go
+++ b/pkg/crypto/keys/private_key.go
@@ -136,8 +136,11 @@ func (p *PrivateKey) Sign(data []byte) []byte {
 	)
 
 	r, s := rfc6979.SignECDSA(privateKey, digest[:], sha256.New)
+	return getSignatureSlice(privateKey.Curve, r, s)
+}
 
-	params := privateKey.Curve.Params()
+func getSignatureSlice(curve elliptic.Curve, r, s *big.Int) []byte {
+	params := curve.Params()
 	curveOrderByteSize := params.P.BitLen() / 8
 	rBytes, sBytes := r.Bytes(), s.Bytes()
 	signature := make([]byte, curveOrderByteSize*2)

--- a/pkg/crypto/keys/sign_verify_test.go
+++ b/pkg/crypto/keys/sign_verify_test.go
@@ -2,6 +2,7 @@ package keys
 
 import (
 	"crypto/ecdsa"
+	"math/big"
 	"testing"
 
 	"github.com/btcsuite/btcd/btcec"
@@ -9,6 +10,34 @@ import (
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 )
+
+func TestIssue1223(t *testing.T) {
+	var d, x, y big.Int
+	d.SetString("75066030006596498716801752450216843918658392116070031536027203512060270094427", 10)
+	x.SetString("56810139335762307690884151098712528235297095596167964448512639328424930082240", 10)
+	y.SetString("108055740278314806025442297642651169427004858252141003070998851291610422839293", 10)
+
+	privateKey := &btcec.PrivateKey{
+		PublicKey: ecdsa.PublicKey{
+			Curve: btcec.S256(),
+			X:     &x,
+			Y:     &y,
+		},
+		D: &d,
+	}
+	pubKey := PublicKey(ecdsa.PublicKey{
+		Curve: btcec.S256(),
+		X:     privateKey.X,
+		Y:     privateKey.Y,
+	})
+
+	hashedData := hash.Sha256([]byte("sample"))
+	signature, err := privateKey.Sign(hashedData.BytesBE())
+	require.NoError(t, err)
+
+	signedData := getSignatureSlice(privateKey.Curve, signature.R, signature.S)
+	require.True(t, pubKey.Verify(signedData, hashedData.BytesBE()))
+}
 
 func TestPubKeyVerify(t *testing.T) {
 	var data = []byte("sample")
@@ -32,7 +61,7 @@ func TestPubKeyVerify(t *testing.T) {
 		assert.Nil(t, err)
 		signature, err := privateKey.Sign(hashedData.BytesBE())
 		require.NoError(t, err)
-		signedData := append(signature.R.Bytes(), signature.S.Bytes()...)
+		signedData := getSignatureSlice(privateKey.Curve, signature.R, signature.S)
 		pubKey := PublicKey(ecdsa.PublicKey{
 			Curve: btcec.S256(),
 			X:     privateKey.X,
@@ -66,7 +95,7 @@ func TestWrongPubKey(t *testing.T) {
 		assert.Nil(t, err)
 		signature, err := privateKey.Sign(hashedData.BytesBE())
 		assert.Nil(t, err)
-		signedData := append(signature.R.Bytes(), signature.S.Bytes()...)
+		signedData := getSignatureSlice(privateKey.Curve, signature.R, signature.S)
 
 		secondPrivKey, err := btcec.NewPrivateKey(btcec.S256())
 		assert.Nil(t, err)

